narciuch/assignment4/src/IslandFinder.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IslandFinder {
+
+  /**
+   * Returns the number of islands in the lands matrix.
+   *
+   * <p>Two tiles in the matrix are part of the same island if
+   * they are both land and are adjacent horizontally or vertically.
+   *
+   * @param nrRows    number of rows in the lands matrix
+   * @param nrColumns number of columns in the lands matrix
+   * @param lands     nested boolean array, with true representing land,
+   *                  false representing water
+   * @return the number of islands in the lands matrix
+   */
+  public int getIslands(int nrRows, int nrColumns, boolean[][] lands) {
+    if (lands.length == 0 || lands[0].length == 0) {
+      return 0;
+    }
+    List<List<Integer>> islands = new ArrayList<>();
```

Comment:
I feel like code could be simplified by using some sort of map, potentially having unique pos as key. Did you consider such approach?